hathor/feature_activation/model/criteria.py

```diff
@@ -58,29 +60,45 @@ class Criteria(BaseModel, validate_all=True):
     lock_in_on_timeout: bool = False
     version: str = Field(..., regex=version.BUILD_VERSION_REGEX)
 
+    def to_validated(self, evaluation_interval: int, max_signal_bits: int) -> 'ValidatedCriteria':
+        """Create a validated version of self, including attribute validations that have external dependencies."""
+        attributes: dict[str, Any] = self.dict() | dict(
```

Comment:
I don't understand why we are using a dict here. Shouldn't we use a typed `NamedTuple` so mypy can check types?